tests/io/vasp/test_sets.py

```diff
@@ -734,68 +732,61 @@ def test_grid_size_from_struct(self):
 
 
 class TestMatPESStaticSet(PymatgenTest):
-    def test_init(self):
-        prev_run = f"{TEST_FILES_DIR}/relaxation"
-
-        vis = MatPESStaticSet.from_prev_calc(prev_calc_dir=prev_run)
-        # Check that INCAR settings were load from yaml, check the default functional is PBE.
-        CONFIG = _load_yaml_config("MatPESStaticSet")
-        assert vis.incar == CONFIG["INCAR"]
-        assert vis.kpoints.style == Kpoints.supported_modes.Monkhorst
-
-        # Check as from dict.
-        vis = MatPESStaticSet.from_dict(vis.as_dict())
-        # Check that INCAR settings were load from yaml.
-        assert vis.incar == CONFIG["INCAR"]
-
-        prev_run_scan = f"{TEST_FILES_DIR}/scan_relaxation"
-
-        vis_scan = MatPESStaticSet.from_prev_calc(prev_calc_dir=prev_run_scan, functional="R2SCAN")
-        # Check that INCAR settings were load from yaml, check the functional "R2SCAN".
-        assert vis_scan.incar["METAGGA"] == "R2SCAN"
-        assert vis_scan.incar["ALGO"] == "ALL"
+    def setUp(self):
+        self.struct = Structure.from_file(f"{TEST_FILES_DIR}/POSCAR")
 
-        # Check as from dict.
-        vis_scan = MatPESStaticSet.from_dict(vis_scan.as_dict())
-        # Check that INCAR settings were load from yaml.
-        assert vis_scan.incar["METAGGA"] == "R2SCAN"
-        assert vis_scan.incar["ALGO"] == "ALL"
+    def test_init_default(self):
+        default = MatPESStaticSet(self.struct)
+        incar = default.incar
+        assert incar["ALGO"] == "Normal"
+        assert incar["EDIFF"] == 1.0e-05
+        assert incar["ENAUG"] == 1360
+        assert incar["ENCUT"] == 680
+        assert incar["GGA"] == "Pe"
+        assert incar["ISMEAR"] == 0
+        assert incar["ISPIN"] == 2
+        assert incar["KSPACING"] == 0.22
+        assert incar["LAECHG"]
+        assert incar["LASPH"]
+        assert incar["LCHARG"]
+        assert incar["LMIXTAU"]
+        assert incar.get("LDAU") is None
+        assert incar["LORBIT"] == 11
+        assert incar["LREAL"] == "Auto"
+        assert not incar["LWAVE"]
+        assert incar["NELM"] == 200
+        assert incar["NSW"] == 0
+        assert incar["PREC"] == "Accurate"
+        assert incar["SIGMA"] == 0.05
 
-        # test with changed user_incar_settings
-        non_prev_vis = MatPESStaticSet(vis.structure, user_incar_settings={"ENCUT": 800, "LORBIT": 12, "LWAVE": True})
-        # Check that the ENCUT and Kpoints style has NOT been inherited.
-        assert non_prev_vis.incar["ENCUT"] == 800
-        # Check that user incar settings are applied.
-        assert non_prev_vis.incar["LORBIT"] == 12
-        assert non_prev_vis.incar["LWAVE"]
+    def test_init_with_prev_incar(self):
```

Comment:
This test is incomplete. Other than checking that the parameters for static are ok, I would want to know that some of the parameters from prev INCAR are moved. Also, an input set also generates KPOINTS and POTCAR files. These need to be checked too, for ALL tests.

Reply:
@shyuep Thanks for the comments. I just saw it because of email address transfer at UCSD. 

I previously have removed all tests for KPOINTS. My thought was that our MatPESStaticSet inherits the KPOINTS related functions directly from DictSet, so there is no need to check KPOINTS here in our tests for MatPESStaticSet. 

I now tends to agree that there is no harm to test more comprehensively, especially as we have KSPACING in INCAR settings.